raml-maven-plugin/src/main/java/uk/gov/justice/raml/maven/lintchecker/rules/LintCheckMojo.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.raml.maven.lintchecker.rules;
+
+import uk.gov.justice.raml.io.FileTreeScanner;
+import uk.gov.justice.raml.io.files.parser.RamlFileParser;
+import uk.gov.justice.raml.maven.common.BasicMojo;
+import uk.gov.justice.raml.maven.lintchecker.processor.LintCheckGoalProcessor;
+import uk.gov.justice.raml.maven.lintchecker.processor.LintCheckerGoalConfig;
+
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "raml-lint-check", requiresDependencyResolution = ResolutionScope.TEST, defaultPhase = LifecyclePhase.GENERATE_SOURCES)
```

Comment:
There is a restriction in the maven-plugin-plugin which we have set, where we specified all goals start with raml-